clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -603,12 +613,10 @@ private void clear() {
                 it.remove();
             }
         }
-        for (String channel : this.failedSends) {
-            KafkaChannel failedChannel = closingChannels.get(channel);
```

Comment:
It seems like this is never possible given that we iterate over the closing channels and remove each from failedSends in the loop above this one.